### DISCUSSION TITLE
Creating nice overlay outputs

### DISCUSSION THREAD

**Discussion by ghost** [Project-MONAI/MONAI#1803] [Q&A] [answered]
Hey,

I'm trying to create easy to digest outputs where the outline of the model segmentation is overlaid with the original image.
At the moment I'm using LabelToContour and some code based off one of the tutorials to create outputs such as:

![image](https://user-images.githubusercontent.com/71658675/111621833-585f5980-87e0-11eb-876a-00766d42bad4.png)

What I'm struggling to do is to get the outline to appear in colour (ie yellow) so that is stands out nicely against the greyscale input image. I've tried changing or removing cmap but my lack of skills is showing because of course this affects the entire combined output eg:

![image](https://user-images.githubusercontent.com/71658675/111621637-20581680-87e0-11eb-9eda-18ed91f0e66f.png)

Does anyone have any advice for how I could best keep the model output/LabelToContour in colour and the input scan in greyscale?

Thanks in advance.

Here is the code:

`      
        plt.subplot(1, 4, 4)

        plt.title(f"Contour overlay: {i}")
        contour = LabelToContour()(largest)
        map_image = contour + val_data["image"]
        plt.imshow(map_image.detach().cpu()[0, 0, :, :, slices_aug[i]], cmap="gray")
        plt.savefig(f'Segmentation {i}.png')
        saver.save_batch(largest)
        plt.show()`

**Comment by rijobro** [comment, marked as answer]
You can set the alpha (opacity) on a voxel-by-voxel basis. Normally, you would probably create another array, with the same shape as `contour` with alpha values between 0 and 1. However, in your case, `contour` already contains the correct alphas! (Because you want `alpha=1` where `contour=1` and 0 elsewhere.)

```python
fig, ax = plt.subplots()
ax.imshow(im, cmap="gray")
ax.imshow(contour, alpha=contour)

ax.set_axis_off()
plt.show()
```

**Reply by rijobro** [reply]
In case it helps you, I created a self contained script:

```python
from glob import glob
import matplotlib.pyplot as plt
import os

from monai.apps import download_and_extract
from monai.transforms import (
    AddChanneld,
    Compose,
    CropForegroundd,
    LabelToContourd,
    Lambdad,
    LoadImaged,
    ToTensord,
)

# get data
directory = os.environ.get("MONAI_DATA_DIRECTORY")
root_dir = tempfile.mkdtemp() if directory is None else os.path.expanduser(directory)
print(root_dir)

task = "Task09_Spleen"
resource = "https://msd-for-monai.s3-us-west-2.amazonaws.com/" + task + ".tar"

compressed_file = os.path.join(root_dir, task + ".tar")
data_dir = os.path.join(root_dir, task)
download_and_extract(resource, compressed_file, root_dir)

# get images
images = sorted(glob(os.path.join(data_dir, "imagesTr", "*.nii.gz")))
labels = sorted(glob(os.path.join(data_dir, "labelsTr", "*.nii.gz")))
data_dicts = [{"image": image, "label": label}
              for image, label in zip(images, labels)]

# transforms 
keys=["image", "label"]
transform = Compose([
    LoadImaged(keys),
    AddChanneld(keys),
    CropForegroundd(keys, source_key="label", margin=[100, 100, 0]),
    ToTensord(keys),
    LabelToContourd("label"),
    Lambdad(keys, lambda x: x[0,...,x.shape[-1]//2]),
])

# get any image
data = transform(data_dicts[1])
im = data["image"]
contour = data["label"]

# plot
fig, ax = plt.subplots()
ax.imshow(im, cmap="gray")
ax.imshow(contour, alpha=contour)

ax.set_axis_off()
plt.show()
```

**Reply by ghost** [reply]
Thank you as ever, Richard! This was perfect and I'm now able to start playing around with many more ways of displaying the results. Thank you!

![image](https://user-images.githubusercontent.com/71658675/111660193-be120c80-8805-11eb-9fb7-d2220e979442.png)

**Reply by Meddebma** [reply]
Hi @jnc13,

I like the vizualization you made! can you share with me the code how you did do the different colours and the model overlay?

thanks!!
